js/timers.js

```diff
@@ -58,10 +92,127 @@ function addTimer() {
+        timerColon.innerHTML=":";
+
+        var timerTimeSec = document.createElement("input");
+        timerTimeSec.classList.add("timerTimeSec");
+        timerTimeSec.type="number";
+        timerTimeSec.value="00";
+        timerTimeSec.max=59;
+
+        timerTime.appendChild(timerTimeMin);
+        timerTime.appendChild(timerColon);
+        timerTime.appendChild(timerTimeSec);
+
+        var addMinuteButton = document.createElement("button");
+        addMinuteButton.classList.add("addMinuteButton");
+        addMinuteButton.innerHTML="<b>+</b> min";
+        addMinuteButton.addEventListener("click", function(){
+            var timerChildren = this.parentNode.childNodes;
+            for(var i = 0; i <timerChildren.length; i++){
+                if(timerChildren[i].classList.contains("timerTime")){
+                    for(var j = 0; j <timerChildren[i].childNodes.length; j++){
+                        var timerTimeChild = timerChildren[i].childNodes[j];
+                        if(timerTimeChild.classList.contains("timerTimeMin")){
+                            timerTimeChild.value = parseInt(timerTimeChild.value)+1;
+                            fireEvent(timerTimeChild,"change");
+                        };
+                    }
+                }
+            }
+        }, false);
+
+        var cancelTimerButton = document.createElement("button");
+        cancelTimerButton.classList.add("cancelTimerButton");
+        cancelTimerButton.innerHTML="Cancel";
+        cancelTimerButton.addEventListener("click", function(){
+            document.getElementById("timers").removeChild(this.parentNode);
+        }, false);
+
+        timerCountdown.id="timer"+currentTimerCount;
+        newtimer.appendChild(timerTitle);
+        newtimer.appendChild(timerTime);
+        newtimer.appendChild(timerCountdown);
+        newtimer.appendChild(addMinuteButton);
+        newtimer.appendChild(cancelTimerButton);
+
+        //newtimer.appendChild(timerContents);
         timers.appendChild(newtimer);
     }
 }
-
+function fireEvent(element,event){
+    if (document.createEventObject){
+        // dispatch for IE
+        var evt = document.createEventObject();
+        return element.fireEvent('on'+event,evt)
+    }
+    else{
+        // dispatch for firefox + others
+        var evt = document.createEvent("HTMLEvents");
+        evt.initEvent(event, true, true ); // event type,bubbling,cancelable
+        return !element.dispatchEvent(evt);
+    }
+}
+function setTimer(newtimer){
+    var time;
+    var minute;
+    var second;
+    var countDown;
+    for (var i = 0; i < newtimer.childNodes.length; i++) {
+        if (newtimer.childNodes[i].className == "timerTime") {
+            time = newtimer.childNodes[i];
+        }
+        else if(newtimer.childNodes[i].className == "timerCountdown") {
+            countDown = newtimer.childNodes[i];
+        }
+    }
+    for (var i = 0; i < time.childNodes.length; i++) {
```

Comment:
you can probably use something like this instead of a for loop 

    this.querySelectorAll("input.timerTimeMin").value = minutes;
    this.querySelectorAll("input.timerTimeSec").value = seconds;